app/controllers/hosts_controller.rb

```diff
@@ -431,6 +432,30 @@ def update_multiple_owner
     redirect_back_or_to hosts_path
   end
 
+  def select_multiple_power_state
+  end
+
+  def update_multiple_power_state
+    # simple validations
+    if (params[:power].nil?) or (action=params["power"]["action"]).nil? or !PowerManager::REAL_ACTIONS.include?(action)
+      error _('No or invalid power state selected!')
+      redirect_to(select_multiple_power_state_hosts_path) and return
+    end
+
+    # set power state for all hosts
+    @hosts.each do |host|
+      begin
+        host.power.send(action.to_sym) if host.supports_power?
+      rescue => error
+        message = _('Failed to set power state for %s.') % host
+        Foreman::Logging.exception(message, error)
+      end
+    end
+
+    notice _('The power state of the selected hosts was set to %s') % _(action)
```

Comment:
Maybe change it to 'will be set to', since by the time this runs, we don't really know if it was set to anything (some call may have failed, some others take some time)